corehq/pillows/case_search.py

```diff
@@ -16,26 +18,47 @@
 from corehq.apps.case_search.exceptions import CaseSearchNotEnabledException
 from corehq.apps.case_search.models import case_search_enabled_domains
 from corehq.apps.change_feed import topics
-from corehq.apps.change_feed.consumer.feed import KafkaChangeFeed, KafkaCheckpointEventHandler
+from corehq.apps.change_feed.consumer.feed import (
+    KafkaChangeFeed,
+    KafkaCheckpointEventHandler,
+)
 from corehq.apps.es import CaseSearchES
 from corehq.elastic import get_es_new
 from corehq.form_processor.backends.sql.dbaccessors import CaseReindexAccessor
 from corehq.form_processor.utils.general import should_use_sql_backend
 from corehq.pillows.mappings.case_mapping import CASE_ES_TYPE
-from corehq.pillows.mappings.case_search_mapping import CASE_SEARCH_INDEX, \
-    CASE_SEARCH_MAPPING, CASE_SEARCH_INDEX_INFO
+from corehq.pillows.mappings.case_search_mapping import (
+    CASE_SEARCH_INDEX,
+    CASE_SEARCH_INDEX_INFO,
+    CASE_SEARCH_MAPPING,
+)
+from corehq.toggles import CASE_LIST_EXPLORER
 from corehq.util.doc_processor.sql import SqlDocumentProvider
 from corehq.util.log import get_traceback_string
 from dimagi.utils.parsing import json_format_datetime
-from pillowtop.checkpoints.manager import get_checkpoint_for_elasticsearch_pillow
+from pillowtop.checkpoints.manager import (
+    get_checkpoint_for_elasticsearch_pillow,
+)
 from pillowtop.es_utils import initialize_index_and_mapping
 from pillowtop.feed.interface import Change
 from pillowtop.pillow.interface import ConstructedPillow
 from pillowtop.processors.elastic import ElasticProcessor
-from pillowtop.reindexer.change_providers.case import get_domain_case_change_provider
-from pillowtop.reindexer.reindexer import PillowChangeProviderReindexer, ReindexerFactory, \
-    ResumableBulkElasticPillowReindexer
-import six
+from pillowtop.reindexer.change_providers.case import (
+    get_domain_case_change_provider,
+)
+from pillowtop.reindexer.reindexer import (
+    PillowChangeProviderReindexer,
+    ReindexerFactory,
+    ResumableBulkElasticPillowReindexer,
+)
+
+
+def domains_needing_search_index():
```

Comment:
this could probably be cached to help the performance on `domain_needs_search_index`

Reply:
good point, will update.